File: app/src/main/java/com/github/angads25/toggledemo/MainActivity.java
package com.github.angads25.toggledemo;

import android.os.Bundle;
import android.graphics.Typeface;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.AppCompatTextView;
import android.support.v4.content.res.ResourcesCompat;

import com.github.angads25.toggle.LabeledSwitch;
import com.github.angads25.toggle.interfaces.OnToggledListener;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        LabeledSwitch labeledSwitch = findViewById(R.id.switch_demo);
        Typeface openSansBold = ResourcesCompat.getFont(MainActivity.this, R.font.open_sans_bold);
        labeledSwitch.setTypeface(openSansBold);
        labeledSwitch.setOnToggledListener(new OnToggledListener() {
            @Override
            public void onSwitched(LabeledSwitch labeledSwitch, boolean isOn) {
                ((AppCompatTextView)findViewById(R.id.label_value))
                        .setText(isOn ? labeledSwitch.getLabelOn() : labeledSwitch.getLabelOff());
            }
        });
    }
}

File: app/src/main/java/com/github/angads25/toggledemo/MainActivity.kt
/*
 * Copyright (C) 2018 Angad Singh
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.github.angads25.toggledemo

import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import android.view.View

class MainActivity : AppCompatActivity(), View.OnClickListener {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        findViewById<View>(R.id.switch_labeled).setOnClickListener(this)
        findViewById<View>(R.id.switch_day_night).setOnClickListener(this)
    }

    override fun onClick(view: View) {
        when (view.id) {
            R.id.switch_labeled -> startActivity(Intent(this@MainActivity, TestActivity::class.java))
            R.id.switch_day_night -> startActivity(Intent(this@MainActivity, DayNightActivity::class.java))
        }
    }
}
